test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+					CPUPolicy:    &noRestart,
+					MemPolicy:    &noRestart,
+					RestartCount: 1,
+				},
+				{
+					Name:         "c3",
+					Resources:    &ContainerResources{CPUReq: "100m", CPULim: "200m", MemReq: "100Mi", MemLim: "200Mi"},
+					CPUPolicy:    &doRestart,
+					MemPolicy:    &doRestart,
+					RestartCount: 1,
+				},
+			},
+		},
+	}
+
+	for idx := range tests {
+		tc := tests[idx]
+		ginkgo.It(tc.name, func(ctx context.Context) {
+			cpuCap, _, _ = getLocalNodeCPUDetails(ctx, f)
+
+			// Skip InPlacePodVerticalScaling tests altogether if the CPU capacity < 2.
+			if cpuCap < 2 {
+				e2eskipper.Skipf("Skipping InPlacePodResize tests since the CPU capacity < 2")
+			}
+
+			var testPod *v1.Pod
+			var patchedPod *v1.Pod
+			var pErr error
+			tStamp := strconv.Itoa(time.Now().Nanosecond())
+			initDefaultResizePolicy(tc.containers)
+			initDefaultResizePolicy(tc.expected)
+			testPod = makeTestPod(f.Namespace.Name, "testpod", tStamp, tc.containers)
+
+			ginkgo.By("creating pod")
+			newPod := podClient.CreateSync(ctx, testPod)
+
+			ginkgo.By("verifying the pod is in kubernetes")
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": tStamp}))
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+			podList, err := podClient.List(context.TODO(), options)
+			framework.ExpectNoError(err, "failed to query for pods")
+			gomega.Expect(podList.Items).Should(gomega.HaveLen(1))
+
+			ginkgo.By("verifying initial pod resources, allocations are as expected")
+			verifyPodResources(newPod, tc.containers)
+			ginkgo.By("verifying initial pod resize policy is as expected")
+			verifyPodResizePolicy(newPod, tc.containers)
+
+			ginkgo.By("verifying initial pod status resources")
+			verifyPodStatusResources(newPod, tc.containers)
+			// Check cgroup values only for containerd versions before 1.6.9
+			if !isInPlaceResizeSupportedByRuntime(f.ClientSet, newPod.Spec.NodeName) {
+				ginkgo.By("verifying pod container's cgroup values before resize")
+				if !framework.NodeOSDistroIs("windows") {
+					verifyPodContainersCgroupValues(ctx, f, newPod, tc.containers, true)
+				}
+			}
+
+			ginkgo.By("patching pod for resize")
+			patchedPod, pErr = f.ClientSet.CoreV1().Pods(newPod.Namespace).Patch(context.TODO(), newPod.Name,
+				types.StrategicMergePatchType, []byte(tc.patchString), metav1.PatchOptions{})
+			framework.ExpectNoError(pErr, "failed to patch pod for resize")
+
+			ginkgo.By("verifying pod patched for resize")
+			verifyPodResources(patchedPod, tc.expected)
+			verifyPodAllocations(patchedPod, tc.containers, true)
+
+			ginkgo.By("waiting for resize to be actuated")
+			resizedPod := waitForPodResizeActuation(ctx, f, f.ClientSet, podClient, newPod, patchedPod, tc.expected)
+
+			// Check cgroup values only for containerd versions before 1.6.9
+			if !isInPlaceResizeSupportedByRuntime(f.ClientSet, newPod.Spec.NodeName) {
```

Comment:
Shouldn't we just skip the test if it's not supported?

Reply:
Done, `supportsInPlacePodVerticalScaling` function handle this now.